### ISSUE TITLE
Improve ecosystem tests compatibility on Windows

### ISSUE THREAD

**Issue by fasttime** [eslint/eslint#20960]
### ESLint version

v10.4.1

### What problem do you want to solve?

Contributors on Windows should be able to run ecosystem tests locally. Support for Windows was intentionally deferred during the initial development of ecosystem tests in #19643 to avoid blocking the rollout.

### What do you think is the correct solution?

When running ecosystem tests locally on Windows, I can see several issues.

#### 1. Symlinks not created during repo cloning

On Windows, regular users are typically not allowed to create symlinks unless developer mode is enabled or elevated privileges are used.
In that situation, `git clone` may check out a symlink as a plain text file instead.

For example, when running ecosystem tests for `eslint-plugin-unicorn` on Windows, the symlink `CLAUDE.md` → `AGENTS.md` is checked out as a plain text file containing the text `AGENTS.md`.
Then, when `npm run test` runs, `markdownlint` reports an error because the file contents do not end with a trailing newline.

```shell
npm run test:ecosystem -- --plugin eslint-plugin-unicorn
```

<details>
<summary>Output</summary>

```shell
> eslint@10.4.1 test:ecosystem
> node tools/test-ecosystem/index.mjs --plugin eslint-plugin-unicorn

Plugins to test: eslint-plugin-unicorn
Clearing existing sandbox directory: ...\eslint\ecosystem

Testing eslint-plugin-unicorn in ...\eslint\ecosystem\eslint-plugin-unicorn
[eslint-plugin-unicorn] git clone https://github.com/sindresorhus/eslint-plugin-unicorn ...\eslint\ecosystem\eslint-plugin-unicorn --depth 1
[eslint-plugin-unicorn] git fetch origin 4fd8dd5aa7c69d397b3d9f83ae66660956bd4c5c
[eslint-plugin-unicorn] git checkout 4fd8dd5aa7c69d397b3d9f83ae66660956bd4c5c
[eslint-plugin-unicorn] cwd: ...\eslint\ecosystem\eslint-plugin-unicorn
[eslint-plugin-unicorn] npm install
[eslint-plugin-unicorn] npm install --no-save ...\eslint
[eslint-plugin-unicorn] npm run test
Failed: eslint-plugin-unicorn

Errors occurred while testing plugins:

eslint-plugin-unicorn: Error: CLAUDE.md:1 error MD041/first-line-heading/first-line-h1 First line in a file should be a top-level heading [Context: "AGENTS.md"]
CLAUDE.md:1:9 error MD047/single-trailing-newline Files should end with a single newline character
ERROR: "lint:markdown" exited with 1.
ERROR: "lint" exited with 1.

    at runCommand (file:///.../eslint/tools/test-ecosystem/index.mjs:69:26)
    at runTests (file:///.../eslint/tools/test-ecosystem/index.mjs:118:2)
    at async file:///.../eslint/tools/test-ecosystem/index.mjs:143:3
```

</details>

Possible mitigations include:

- removing problematic symlink files before running tests
- excluding such files from `markdownlint`
- cloning with `--config core.symlinks=true` so git fails fast when symlinks cannot be created

#### 2. `pnpm install` hanging indefinitely for `typescript-eslint`

When running:

```shell
npm run test:ecosystem -- --plugin @typescript-eslint/typescript-eslint
```

the process appears to stop during `pnpm install`, seemingly waiting for stdin input. I was able to fix this locally by setting `CI=true` in the environment variables.
Maybe there are other ways to make `pnpm` run in non-interactive mode in ecosystem tests.

#### 3. `pnpm link` fails when the path contains spaces

If the local ESLint checkout is in a path containing spaces, for example `C:\Users\John Doe\eslint`, ecosystem tests for `@typescript-eslint/typescript-eslint` fail during `pnpm link`:

```shell
npm run test:ecosystem -- --plugin @typescript-eslint/typescript-eslint
```

<details>
<summary>Output</summary>

```shell
> eslint@10.4.1 test:ecosystem
> node tools/test-ecosystem/index.mjs --plugin @typescript-eslint/typescript-eslint

Plugins to test: @typescript-eslint/typescript-eslint
Clearing existing sandbox directory: C:\Users\John Doe\eslint\ecosystem

Testing @typescript-eslint/typescript-eslint in C:\Users\John Doe\eslint\ecosystem\typescript-eslint-typescript-eslint
[@typescript-eslint/typescript-eslint] git clone https://github.com/typescript-eslint/typescript-eslint C:\Users\John Doe\eslint\ecosystem\typescript-eslint-typescript-eslint --depth 1 --config core.symlinks=true
[@typescript-eslint/typescript-eslint] git fetch origin 5cf1b1f61e17943c7b598e779526e18ccac4ccd5
[@typescript-eslint/typescript-eslint] git checkout 5cf1b1f61e17943c7b598e779526e18ccac4ccd5
[@typescript-eslint/typescript-eslint] cwd: C:\Users\John Doe\eslint\ecosystem\typescript-eslint-typescript-eslint
[@typescript-eslint/typescript-eslint] pnpm install
[@typescript-eslint/typescript-eslint] pnpm link C:\Users\John Doe\eslint
Failed: @typescript-eslint/typescript-eslint

Errors occurred while testing plugins:

@typescript-eslint/typescript-eslint: Error: 'C:\Users\John' is not recognized as an internal or external command,
operable program or batch file.

    at runCommand (file:///C:/Users/John%20Doe/eslint/tools/test-ecosystem/index.mjs:70:26)
    at runTests (file:///C:/Users/John%20Doe/eslint/tools/test-ecosystem/index.mjs:112:3)
    at async file:///C:/Users/John%20Doe/eslint/tools/test-ecosystem/index.mjs:146:3
```

</details>

This may be a bug in `pnpm`, but it could be likely avoided by not passing an absolute path with spaces to `pnpm link`.
For example, using a relative path (`pnpm link ../..`) instead of the absolute path `C:\Users\John Doe\eslint`.

### Participation

- [ ] I am willing to submit a pull request for this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by crimsonjay0** [comment]
I have a Windows machine, so I can take a look at this issue.

**Comment by DMartens** [comment]
Feel free to open a PR for this @crimsonjay0.

**Comment by lumirlumir** [comment]
@crimsonjay0 Are you still available to work on it?
